opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -79,23 +79,27 @@ message ScopeSpans {
 // The next available field id is 17.
 message Span {
   // A unique identifier for a trace. All spans from the same trace share
-  // the same `trace_id`. The ID is a 16-byte array. An ID with all zeroes
-  // is considered invalid.
-  //
-  // This field is semantically required. Receiver should generate new
-  // random trace_id if empty or invalid trace_id was received.
+  // the same `trace_id`. The ID is a 16-byte array. An ID with all zeroes OR
+  // of length other than 16 bytes is considered invalid (empty string in OTLP/JSON
+  // is zero-length and thus is also invalid).
   //
   // This field is required.
+  //
+  // The receivers SHOULD generate a new random trace_id if any of the following is true:
+  //   - the field is not present,
+  //   - the field contains an invalid value.
   bytes trace_id = 1;
 
   // A unique identifier for a span within a trace, assigned when the span
-  // is created. The ID is an 8-byte array. An ID with all zeroes is considered
-  // invalid.
-  //
-  // This field is semantically required. Receiver should generate new
-  // random span_id if empty or invalid span_id was received.
+  // is created. The ID is an 8-byte array. An ID with all zeroes OR of length
+  // other than 8 bytes is considered invalid (empty string in OTLP/JSON
+  // is zero-length and thus is also invalid).
   //
   // This field is required.
+  //
+  // The receivers SHOULD generate a new random span_id if any of the following is true:
```

Comment:
Also disagree with generated spans here.  This would break log<->span and span<->exemplar correlation.  THis should just treat the span as invalid if the id is invalid.

Reply:
https://github.com/open-telemetry/opentelemetry-proto/pull/444 is now merged and we no longer generate random ids.